src/playlist.rs

```diff
@@ -1144,6 +1208,222 @@ impl DateRange {
     }
 }
 
+// Implementing structs for LL-HLS
+#[derive(Debug, Default, PartialEq, Clone)]
+pub struct ServerControl {
+    pub can_skip_until: Option<f64>,
+    pub hold_back: Option<f64>,
+    pub part_hold_back: Option<f64>,
+    pub can_block_reload: bool,
+}
+
+impl ServerControl {
+    pub(crate) fn from_hashmap(
+        mut attrs: HashMap<String, QuotedOrUnquoted>,
+    ) -> Result<ServerControl, String> {
+        let can_skip_until = unquoted_string_parse!(attrs, "CAN-SKIP-UNTIL", |s: &str| s
+            .parse::<f64>()
+            .map_err(|err| format!("Failed to parse CAN-SKIP-UNTIL attribute: {}", err)));
+        let hold_back = unquoted_string_parse!(attrs, "HOLD-BACK", |s: &str| s
+            .parse::<f64>()
+            .map_err(|err| format!("Failed to parse HOLD-BACK attribute: {}", err)));
+        let part_hold_back = unquoted_string_parse!(attrs, "PART-HOLD-BACK", |s: &str| s
+            .parse::<f64>()
+            .map_err(|err| format!("Failed to parse PART-HOLD-BACK attribute: {}", err)));
+        let can_block_reload = is_yes!(attrs, "CAN-BLOCK-RELOAD");
+
+        Ok(ServerControl {
+            can_skip_until,
+            hold_back,
+            part_hold_back,
+            can_block_reload,
+        })
+    }
+
+    pub(crate) fn write_to<T: Write>(&self, w: &mut T) -> std::io::Result<()> {
+        write!(w, "#EXT-X-SERVER-CONTROL:")?;
+        write_some_attribute!(w, "CAN-SKIP-UNTIL", &self.can_skip_until)?;
+        write_some_attribute!(w, ",HOLD-BACK", &self.hold_back)?;
+        write_some_attribute!(w, ",PART-HOLD-BACK", &self.part_hold_back)?;
+        if self.can_block_reload {
+            write!(w, ",CAN-BLOCK-RELOAD=YES")?;
+        }
+        writeln!(w)
+    }
+}
+
+#[derive(Debug, Default, PartialEq, Clone)]
+pub struct PartInf {
+    pub part_target: f64,
+}
+
+impl PartInf {
+    pub(crate) fn from_hashmap(
+        mut attrs: HashMap<String, QuotedOrUnquoted>,
+    ) -> Result<PartInf, String> {
+        let part_target = unquoted_string_parse!(attrs, "PART-TARGET", |s: &str| s
+            .parse::<f64>()
+            .map_err(|err| format!("Failed to parse PART-TARGET attribute: {}", err)))
+        .ok_or_else(|| String::from("EXT-X-PART-INF without mandatory PART-TARGET attribute"))?;
+
+        Ok(PartInf { part_target })
+    }
+
+    pub(crate) fn write_to<T: Write>(&self, w: &mut T) -> std::io::Result<()> {
+        write!(w, "#EXT-X-PART-INF:PART-TARGET={}", self.part_target)?;
```

Comment:
I believe everywhere where a `f64` is written out, it needs to use the same if/else construction as in [this](https://github.com/rutgersc/m3u8-rs/pull/72/files#diff-294648967369087cf3619793bd50043a2b39b5f481523d7c1636cc3e8d572a5bR891-R899) example. Some parsers cannot deal with floats when they don't have decimals, so it's good to apply it to all the new `f64` fields so it's consistent with the rest.